src/test/rpc/RPCCall_test.cpp

```diff
@@ -0,0 +1,7037 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2018 Ripple Labs Inc.
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+
+#include <ripple/net/RPCCall.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/json/json_reader.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <test/jtx.h>
+
+#include <initializer_list>
+#include <vector>
+
+namespace ripple {
+namespace test {
+
+struct RPCCallTestData
+{
+    char const* const description;
+    int line;
+    // List of passed arguments.
+    std::initializer_list<char const* const> const args;
+
+    // If it throws, what does it throw?
+    enum Exception
+    {
+        no_exception = 0,
+        bad_cast
+    };
+    Exception throwsWhat;
+
+    // Expected JSON response.
+    char const* const exp;
+
+    RPCCallTestData () = delete;
+    RPCCallTestData (RPCCallTestData const&) = delete;
+    RPCCallTestData (RPCCallTestData&&) = delete;
+    RPCCallTestData& operator= (RPCCallTestData const&) = delete;
+    RPCCallTestData& operator= (RPCCallTestData&&) = delete;
+};
```

Comment:
In C++20 `RPCCallTestData` will no longer be an aggregate type because of the user-declared constructors.  We should fix this now.  One way would be to add the constructor used in `rpcCallTestArray`.  Another way is to remove these deleted special members.

Reply:
A real improvement to the language, by the way.

I added the constructor, so we'll be ready for C++20.  At least here.